### DISCUSSION TITLE
New Cache System and Data Pipeline

### DISCUSSION THREAD

**Discussion by iangilman** [openseadragon/openseadragon#2637] [General]
We've just merged a big upgrade to the OSD data cache and pipeline: #2407. We're excited about the new functionality, but we want to make sure it's good and solid. We'd love to get as many people as possible trying it out before we release it!

These changes are particularly of interest to anyone who modifies the image or uses unusual data for their image. If your project uses the OpenSeadragonFiltering plugin or similar, you'll definitely want to check it out. But we're also interested in testing from people who aren't doing anything special, just to make sure it works in all scenarios. 

To try it out, you'll need to build from the master branch of this repository; see the [contributing guide](https://github.com/openseadragon/openseadragon/blob/master/CONTRIBUTING.md) for instructions. 

To learn about the new functionality, see the 6.0.0 section of the [changelog](https://github.com/openseadragon/openseadragon/blob/master/changelog.txt). There's also a lot of new documentation over in this PR: https://github.com/openseadragon/site-build/pull/289 but it's not so easy to read in that format; maybe we should drop some of it in the wiki or something for the time being until we're ready to merge it. 

Anyway, please give it a try! We're excited for this new chapter 😊

**Comment by Aiosa** [comment]
I guess we should then try to update the plugin list, and check that they still work - and fix, or remove them. I will release the filtering plugin update which uses the new API.

**Reply by iangilman** [reply]
Sounds great!

**Comment by iangilman** [comment]
Here's a built version of the current master, in case you want to try it but aren't set up to build it yourself. It's better to build directly off of master, because then you can keep up-to-date with changes, but this is a reasonable starting point.

[openseadragon-bin-6.0.0a1.zip](https://github.com/user-attachments/files/18169280/openseadragon-bin-6.0.0a1.zip)

**Comment by iangilman** [comment]
Here are the new data modification docs:

![OSD data modification docs](https://github.com/user-attachments/assets/d9edc041-95fa-4d74-b730-7cf10951d0a9)

**Comment by Aiosa** [comment]
The new filtering plugin is available here: https://github.com/RationAI/OpenSeadragonFiltering

**Reply by iangilman** [reply]
Is this going to be the new official one? Sounds good to me. I guess we can update https://github.com/openseadragon/site-build/pull/289 to include it in the plugins list? We should probably keep the old one as well, for people who are using earlier versions of OSD.

**Reply by Aiosa** [reply]
I hope so. We could create a new list where plugins that are confirmed to work are listed, and slowly update.

**Reply by iangilman** [reply]
I image most plugins won't be affected by the cache overhaul, but yes, it would be good to keep track of which ones work and which ones don't.
